### PR TITLE
add mobx v2.1.4 with npm autoupdate config

### LINKED ISSUE #7697: [Request] Add MobX
**Library name: MobX** 
**Git repository url: https://github.com/mobxjs/mobx**
**npm package url: https://www.npmjs.com/package/mobx** 
**License(s): MIT**
**Official homepage: http://mobxjs.github.io/mobx/index.html**

### PR DESCRIPTION
**Pull request by dannyxx001**
PR for #7697 
@pvnr0082t please help me review this PR, thanks.

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/mobxjs/mobx
 * Official website (optional, not the repository): http://mobxjs.github.io/mobx/index.html
 * NPM package url (optional): https://www.npmjs.com/package/mobx
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 1600
   - Count of watchers: 72
   - Count of forks: 66
 * NPM download stats (optional):
   - Downloads in the last day: 155
   - Downloads in the last week: 930
   - Downloads in the last month: 3521

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #7697, cc @mweststrate

### CODE REVIEW

**Comment by pvnr0082t** [comment]
@PeterDaveHello What's the ```.ts``` file do? Does cdnjs host ```.ts``` file?

**Comment by mweststrate** [comment]
@pvnr0082t  The `.ts` file is for develop / build time only and supplies the type information of the package. I don't think it adds anything to host this on a CDN.

**Comment by PeterDaveHello** [comment]
@pvnr0082t : As @mweststrate says, we don't need that on CDN, should remove it.
Thanks @mweststrate .

**Comment by dannyxx001** [comment]
OK, I had updated it. @pvnr0082t please review again, thanks.

File: ajax/libs/mobx/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "model",
+    "frp",
+    "functional-reactive-programming",
+    "state management",
+    "data flow"
+  ],
+  "npmName": "mobx",
+  "npmFileMap": [
+    {
+      "basePath": "",
```
Comment (PeterDaveHello) [inline ajax/libs/mobx/package.json]: Should use `lib` as basePath here.

Comment (dannyxx001) [inline ajax/libs/mobx/package.json]: OK, I will remember it next time.

**Pull request merged**